services/AttendanceService.ts

```diff
@@ -63,22 +63,30 @@ export default class AttendanceService {
     return Repositories.attendance(txn).attendEvent(user, event, attendedAsStaff);
   }
 
-  public async submitAttendanceForUser(userUuid: Uuid, eventUuid: Uuid, asStaff = false,
-    admin: UserModel): Promise<PublicAttendance> {
+  public async submitAttendanceForUsers(emails: string[], eventUuid: Uuid, asStaff = false,
+    admin: UserModel): Promise<PublicAttendance[]> {
     return this.transactions.readWrite(async (txn) => {
       const event = await Repositories.event(txn).findByUuid(eventUuid);
       if (!event) throw new NotFoundError('This event doesn\'t exist');
 
-      const user = await Repositories.user(txn).findByUuid(userUuid);
-      if (!user) throw new NotFoundError('This user was not found');
+      const users = await Repositories.user(txn).findByEmails(emails);
 
-      const attendanceRepository = Repositories.attendance(txn);
-      const hasAlreadyAttended = await attendanceRepository.hasUserAttendedEvent(user, event);
-      if (hasAlreadyAttended) throw new UserError('This user has already attended this event');
+      if (users.length !== emails.length) {
+        throw new BadRequestError('Couldn\'t find accounts matching one or more emails');
+      }
 
-      const activityDescription = `Attendance submitted by user ${admin.uuid}`;
-      const attendance = await this.writeEventAttendance(user, event, asStaff, txn, activityDescription);
-      return attendance.getPublicAttendance();
+      return Promise.all(users.map(async (user) => {
+        const attendanceRepository = Repositories.attendance(txn);
+        const hasAlreadyAttendedEvent = await attendanceRepository.hasUserAttendedEvent(user, event);
+        if (hasAlreadyAttendedEvent) {
```

Comment:
Looking back I think it's probably better to just skip instead of throwing, since the service should just flag everyone as attended, and if they already attended, then no work needs to be done for that user